shells/test/utils.js

```diff
@@ -8,138 +8,67 @@
-};
-
-exports.keys = async function(selector, keys, timeout) {
-  await exports.click(selector, timeout);
-  await browser.keys(keys);
+exports.keys = function(selector, keys, timeout) {
+  browser.$(selector).click();
+  browser.keys(keys);
 };
 
 /**
- * Start a new arc in the webdriver environment.
- * @param storage pouchdb or firebase
+ * Start a new arc in the webdriver environment.  Performs the following
+ * steps:
+ * - Calculates a persona based on the storgeType
+ * - Opens a new window to the web shell
+ * - Moves focus to the search bar so it pops upwards.
+ * - Waits for animations
+ * - Clicks on the search box.
+ * - Pauses 1s.
+ *
+ * @param testTitle Displayed in logs
+ * @param storage the storage type to test firebase, firebasetest, pouchdb.
  */
+exports.openNewArc = function(testTitle, storageType) {
+  const user = 'selenium-' + Date.now();  // unique enough for now.
 
-exports.openNewArc = async function(testTitle, useSolo, storageType) {
-  // clean up extra open tabs
-  const openTabs = browser.getTabIds();
-  browser.switchTab(openTabs[0]);
-  openTabs.slice(1).forEach(tabToClose => {
-    browser.close(tabToClose);
-  });
-
-  const storageKey = storageKeyByType[storageType];
-  let storage;
-  let suffix;
-
-  switch (storageType) {
-
-  case 'pouchdb':
-    // setup url params
-    suffix = `${Date.now()}-${testTitle}`.replace(/\W+/g, '-').replace(/\./g, '_');
-    storage = `${storageKey}/${suffix}/`;
-    break;
-  case 'firebase':
-    suffix = `${new Date().toISOString()}_${testTitle}`.replace(/\W+/g, '-').replace(/\./g, '_');
-    storage = `${storageKey}/${suffix}`;
-    //console.log(`running test "${testTitle}" with firebaseKey "${firebaseKey}"`);
-    break;
-  default:
-    throw new Error('must specify firebase/pouchdb parameter');
-  }
-  console.log(`running test "${testTitle}" [${storage}]`);
+  console.log(`running test "${testTitle}" [${storageType}]`);
   const urlParams = [
-    //`testFirebaseKey=${firebaseKey}`,
-    //`log`,
+    `log`,
     `plannerStorage=volatile`,
-    `storage=${storage}`,
-    'user=selenium'
+    `persona=$${storageType}/${user}`
   ];
-  if (useSolo) {
-    urlParams.push(`solo=${browser.options.baseUrl}/artifacts/canonical.manifest`);
-  }
+
   // note - baseUrl (currently specified on the command line) must end in a
   // trailing `/`, and this must not begin with a preceding `/`.
   // `browser.url()` will prefix its argument with baseUrl, and avoiding a
   // doubling `//` situation avoids some bugs.
-  await browser.url(`${exports.shellUrl}/?${urlParams.join('&')}`);
-  // only to ensure time for the app to configure itself
-  await exports.waitFor('input[search]');
+  browser.url(`${exports.shellUrl}/?${urlParams.join('&')}`);
+
+  const panelUi =  $('//web-shell/web-shell-ui').shadow$('system-ui').shadow$('panel-ui');
+  const searchBar = panelUi.shadow$('#searchinput');
+  searchBar.moveTo(0, 0);  // need 0,0 because mid-point is off screen.
```

Comment:
You've changed the semantics of the tests. :)

IMO, this 'testing by position' thing should only be used very specifically for UI tests. For integration tests (which these are), position testing is brittle, can get in trouble due to animations and so on that would not trip up an actual human. I suspect this is why there is 'wait a little' and friends below.

The infrastructure I had previously avoided some of this brittle webdriver API in favor of working with nodes directly (you click a node, not a position where you expect the node to be).

Reply:
indeed, this is what needs to get back.  The undocumented json that was used seems to have changed in the new version; and I haven't had time to track down the dark arts to bring it back.